### DISCUSSION TITLE
《人民日报》爬虫

### DISCUSSION THREAD

**Discussion by PirateBook1** [banned-historical-archives/banned-historical-archives.github.io#2153] [General]
@CiangCing14
请求帮助制作爬虫解析《人民日报》数据
有两处数据源：
https://www.laoziliao.net/rmrb/
https://cn.govopendata.com/renminribao/

就数据源数据质量来看，解析出的单篇文稿只要求三个字段：title、date 和 content，不需要 author（难以解析）

@banned-historical-archives

**Comment by banned-historical-archives** [comment]
https://github.com/banned-historical-archives/rmrb

**Reply by banned-historical-archives** [reply]
大部分都不是敏感内容，数量又太大，讨论一下是否有必要录入

**Reply by banned-historical-archives** [reply]
可以挑一些年份比如49-76

**Reply by banned-historical-archives** [reply]
即使这样，目测也有几十万篇文章，很影响文稿列表页面加载速度

**Reply by PirateBook1** [reply]
两报一刊还是完全收录吧，收到 76 年就好，目前的数据量已经开始影响性能了吗

**Reply by PirateBook1** [reply]
可以想一些折中的办法

**Reply by banned-historical-archives** [reply]
现在差不多两万篇 页面2mb的样子

**Reply by PirateBook1** [reply]
或者看看还有没有性能优化的空间，数据压缩什么的

**Reply by banned-historical-archives** [reply]
可以加个按钮，列出包括人民日报的所有文稿

**Comment by PirateBook1** [comment]
@banned-historical-archives 1946-1976.10 解析出了 1.44 GB 的数据，五十多万篇，如何处理

**Reply by banned-historical-archives** [reply]
过段时间我来录入吧，你先录别的

**Comment by banned-historical-archives** [comment]
这个数据量构建肯定超过6小时了，只能在本地部署时解析人民日报

**Reply by PirateBook1** [reply]
@banned-historical-archives 对于本地部署时解析，有没有一些流程化步骤可以供我参考（在已经构建好现成文本对象的情况下），我想录一些网络材料

**Reply by PirateBook1** [reply]
郭建波，《伟大的理论 光辉的实践》

**Reply by banned-historical-archives** [reply]
@PirateBook1 不懂

**Reply by PirateBook1** [reply]
@banned-historical-archives 就是比如我有一个新的 xxx.ts，怎么录进数据库，像 GPCRv3 那样

**Reply by banned-historical-archives** [reply]
创建 

books/GPCRv3.ts
parser/GPCR_v3/index.ts

books.ts 增加
```
import * as GPCRv3 from './parser/GPCR_v3';

const parsers: { [key: string]: any } = { // 在这里注册新的解析器
  jinghuo,
  wansui,
  wenji,
  GPCRv3,
  jqjianghua,
  xuanji,
  jimi,
  wenku,
  maoquanji,
  wanghongwen,
  yaowenyuan,
  zhangchunqiao,
  zzj1,
  automation,
};
```

**Comment by PirateBook1** [comment]
@banned-historical-archives 发现一些四人的讲话，在人民日报里有但是目前没有收录

**Reply by banned-historical-archives** [reply]
晚点我来录入

**Comment by banned-historical-archives** [comment]
已录入
